pkg/virt-controller/watch/vmi.go

```diff
@@ -1279,6 +1272,105 @@ func (c *VMIController) sync(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod,
 	return nil
 }
 
+func (c *VMIController) handleHotplugVolumes(hotplugVolumes []*virtv1.Volume, hotplugAttachmentPods []*k8sv1.Pod, vmi *virtv1.VirtualMachineInstance, virtLauncherPod *k8sv1.Pod, dataVolumes []*cdiv1.DataVolume) error {
+	logger := log.Log.Object(vmi)
+	key, err := controller.KeyFunc(vmi)
+	if err != nil {
+		logger.Object(vmi).Reason(err).Error("failed to extract key from virtualmachine.")
+		return fmt.Errorf("failed to extract key from virtualmachine. %v", err)
+	}
+	// Only consider pods which belong to this vmi
+	// excluding unfinalized migration targets from this list.
+	virtlauncherPod, err := controller.CurrentVMIPod(vmi, c.podInformer)
+	if err != nil {
+		logger.Reason(err).Error("Failed to fetch pods for namespace from cache.")
+		return err
+	}
+
+	readyHotplugVolumes, err := c.getReadyHotplugVolumes(hotplugVolumes, vmi, dataVolumes, virtlauncherPod)
+	if err != nil {
+		return err
+	}
+	logger.V(4).Infof("Number of ready volumes %d", len(readyHotplugVolumes))
+
+	currentPod, oldPods := c.getCurrentAndOldAttachmentPods(readyHotplugVolumes, hotplugAttachmentPods)
+
+	if len(readyHotplugVolumes) > 0 {
+		if !c.requeueAfter(currentPod, key, c.requeTimeByVolumes(len(hotplugVolumes))*time.Second) {
+			if err := c.createAttachmentPodIfNotRunning(vmi, currentPod, oldPods, virtlauncherPod, readyHotplugVolumes); err != nil {
+				return err
+			}
```

Comment:
this part looks great.

However there are a bunch of changes unrelated to this change throughout this PR. I understood the purpose of those changes when the goal was to utilize a new queue. now without that justification, this looks like a refactor of code mixed with a subtle change in logic.